subworkflows/local/busco_diamond_blastp.nf

```diff
@@ -0,0 +1,99 @@
+//
+//
+// Run BUSCO for a genome from GOAT and runs diamond_blastp
+//
+//
+
+nextflow.enable.dsl = 2
+
+include { GOAT_TAXONSEARCH    } from '../../modules/local/goat_taxon_search'
+include { BUSCO               } from '../../modules/nf-core/modules/busco/main'
+//include { EXTRACT_BUSCO_GENES } from '../../modules/local/extract_busco_genes'
+//include { DIAMOND_BLASTP      } from '../../modules/nf-core/modules/diamond/blastp/main'
+
+workflow BUSCO_DIAMOND {
+    take:
+    //  Path to genome fasta file:
+    fasta
+
+    // GOAT_TAXONSEARCH input
+    /// Value: single binomial name or NCBI taxonomy ID or '' if taxa_file is provided
+    taxon
+    /// File containing a taxon ID per line or empty list if taxon is provided
+    taxa_file
+
+    // DIAMOND_BLASTP input
+    /// Directory containing the protein blast database:
+    //diamonddb
+    /// Specify the type of output file to be generated, `txt` corresponds to to BLAST tabular format:
+    //outext
+    /// Space separated list of DIAMOND tabular BLAST output keywords:
+    /// "qseqid staxids bitscore qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore"
+    //blast_cols
+
+    main:
+
+    ch_versions = Channel.empty()
+
+    // this is the string used to name all intermediate and final output files
+    name = fasta.map { f -> f.simpleName }
+
+
+    //
+    // Fetch BUSCO lineages for taxon (or taxa)
+    //
+
+    GOAT_TAXONSEARCH (
+        name,
+        taxon,
+        taxa_file // input: taxon, no taxa_file or no taxon, taxa_file
```

Comment:
I always find it tricky to combine multiple channels from `take:`. Here, personally, I wouldn't pass the taxon information as inputs of the subworkflows, and directly using `params`, e.g.
```suggestion
        fasta.map { fa -> [fa[0], params.taxon, params.taxa_file ? file(params.taxa_file) : []] },
```
and that's the tuple you give goat/taxonsearch

Reply:
Done. There will be also two commits: in one I just removed these parameters from `workflows/blobtoolkit.nf` and other one to remove them from the subworkflow script.

Reply:
No please don't do this. Passing params directly is considered bad practice in nf-core, I had this in one of my subworkflows and they recommended I create a channel and only pass those. Passing params directly sometimes leads to unexpected behaviour. I am happy to discuss this further, but everything should be passed as a channel. The only place params should be directly referenced is the main workflow, when you create appropriate channels.